### PR TITLE
fix(medusa) : Improperly referenced product id in product import strategy

### PR DESCRIPTION
**Pull request by ayushthe1**
* What - fixing PR #2350 

* Why - In the ProductImportStrategy class, the getImportInstructions improperly checks the product.id for each row. With it setup this way, products will always be considered new because the product.id column is not read here. So products will never be updated. [Link to the code line](https://github.com/medusajs/medusa/blob/bd941309161bcee4fbfc7918d310c426ba8d8f15/packages/medusa/src/strategies/batch-jobs/product/import.ts#L166)

* How - Replacing `row["product.product.id"]` to  `row["product.id"]`

### CODE REVIEW

**Comment by ayushthe1** [comment]
@shahednasser Could you please review this pr .

**Comment by shahednasser** [comment]
@ayushthe1 I only review documentation-related PRs. Please wait while someone from the engineering team gets the chance to review it 😊

**Comment by ayushthe1** [comment]
> @ayushthe1 I only review documentation-related PRs. Please wait while someone from the engineering team gets the chance to review it 😊

Thanks .. sorry for bugging you 😅.

File: packages/medusa/src/strategies/__tests__/batch-jobs/product/import.ts [hunk 1]
```diff
@@ -35,7 +35,8 @@ async function* generateCSVDataForStream() {
...
-  yield "5VxiEkmnPV,test-product-product-2,Test product,,test-product-description,draft,,,,,,,,,,Test collection,test-collection2,test-type,123,TRUE,,profile_2,profile_type_2,3SS1MHGDEJ,Test variant,test-sku-3,test-barcode-3,10,FALSE,TRUE,,,,,,,,,,120,,,test-option,Option 1 Value blue,,,test-image.png\n"
+  yield "5VxiEkmnPV,test-product-product-3,Test product,,test-product-description,draft,,,,,,,,,,Test collection,test-collection2,test-type,123,TRUE,,profile_2,profile_type_2,3SS1MHGDEJ,Test variant,test-sku-3,test-barcode-3,10,FALSE,TRUE,,,,,,,,,,120,,,test-option,Option 1 Value blue,,,test-image.png\n"
+  yield ",test-product-product-4,Test product,,test-product-description,draft,,,,,,,,,,Test collection,test-collection2,test-type,123,TRUE,,profile_2,profile_type_2,3SS1MHGDEJ,Test variant,test-sku-3,test-barcode-3,10,FALSE,TRUE,,,,,,,,,,120,,,test-option,Option 1 Value blue,,,test-image.png\n"
```
Comment (olivermrbl) [inline packages/medusa/src/strategies/__tests__/batch-jobs/product/import.ts:39]: @fPolic As far as I could see, we did not have a test for creating a product in an import. Can I get you to validate? I've added an import line for a product without id

**Review by fPolic — APPROVED** [review]
LGTM @ayushthe1! Appreciate the contribution 🚀

File: packages/medusa/src/strategies/batch-jobs/product/utils.ts [hunk 2]
```diff
@@ -54,12 +56,14 @@ export function transformVariantData(
 
   Object.keys(productData).forEach((k) => {
     const key = k.split("variant.")[1]
+
+    set(ret, key, productData[k])
```
Comment (olivermrbl) [inline packages/medusa/src/strategies/batch-jobs/product/utils.ts:60]: question: Line 60 + 61 are duplicates. We should be able to safely delete one of them no?

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**